athena/data/feature_normalizer.py

```diff
@@ -22,17 +22,54 @@
 import pandas
 from absl import logging
 import tensorflow as tf
+import numpy as np
+import multiprocessing as mp
 
+def tqdm_listener(q, total):
+    pbar = tqdm.tqdm(total = total)
+    for item in iter(q.get, None):
+        pbar.update()
+
+def compute_cmvn_by_chunk(feature_dim, tar_speaker, featurizer, entries, msg_queue):
+    initial_mean = tf.Variable(tf.zeros([feature_dim], dtype=tf.float32))
+    initial_var = tf.Variable(tf.zeros([feature_dim], dtype=tf.float32))
+    total_num = tf.Variable(0, dtype=tf.int32)
+
+    if msg_queue is None:
+        entries = tqdm.tqdm(entries)
+
+    for items in entries:
+        audio_file, speaker = items[0], items[-1]
+        if speaker != tar_speaker:
+            continue
+        feat_data = featurizer(audio_file)
+        temp_frame_num = feat_data.shape[0]
+        total_num.assign_add(temp_frame_num)
+
+        temp_feat = tf.reshape(feat_data, [-1, feature_dim])
+        temp_feat2 = tf.square(temp_feat)
+
+        temp_mean = tf.reduce_sum(temp_feat, axis=[0])
+        temp_var = tf.reduce_sum(temp_feat2, axis=[0])
+
+        initial_mean.assign_add(temp_mean)
+        initial_var.assign_add(temp_var)
+        if msg_queue is not None:
+            msg_queue.put(1)
+    return initial_mean, initial_var, total_num
 
 class FeatureNormalizer:
     """ Feature Normalizer """
 
-    def __init__(self, cmvn_file=None):
+    def __init__(self, cmvn_config=None):
```

Comment:
Better not modify init param, because this class is public used, it will be call in many other position, we can pass param(num_core) in compute_cmvn function.

Reply:
sure